class-acf-field-svg-icon-picker.php

```diff
@@ -0,0 +1,156 @@
+
+		/**
+		 * Controls field type visibility in REST requests.
+		 *
+		 * @var bool
+		 */
+		public $show_in_rest = true;
+
+		/**
+		 * Stores the path suffix to the icons
+		 *
+		 * @var string
+		 */
+		private string $path_suffix;
+
+		/**
+		 * Stores the path to the icons
+		 *
+		 * @var string
+		 */
+		private string $path;
+
+		/**
+		 * Stores the url to the icons
+		 *
+		 * @var string
+		 */
+		private string $url;
+
+		/**
+		 * Stores the icons
+		 *
+		 * @var array
+		 */
+		private array $svgs = array();
+
+
+		/**
+		 * Constructor.
+		 *
+		 * We set the field name, label, category, defaults and l10n.
+		 */
+		public function __construct() {
+			$this->name        = 'icon-picker';
+			$this->label       = __( 'SVG Icon Picker', 'acf-svg-icon-picker' );
+			$this->category    = 'content';
+			$this->defaults    = array( 'initial_value' => '' );
+			$this->l10n        = array( 'error' => __( 'Error!', 'acf-svg-icon-picker' ) );
+			$this->path_suffix = apply_filters( 'acf_icon_path_suffix', 'assets/img/acf/' );
+			$this->path        = apply_filters( 'acf_icon_path', acf_plugin_svg_icon_picker::$settings['path'] ) . $this->path_suffix;
+			$this->url         = apply_filters( 'acf_icon_url', acf_plugin_svg_icon_picker::$settings['url'] ) . $this->path_suffix;
+
+			$priority_dir_lookup = get_stylesheet_directory() . '/' . $this->path_suffix;
+
+			if ( file_exists( $priority_dir_lookup ) ) {
+				$this->path = $priority_dir_lookup;
+				$this->url  = get_stylesheet_directory_uri() . '/' . $this->path_suffix;
+			}
+
+			$files = array_diff( scandir( $this->path ), array( '.', '..' ) );
+			foreach ( $files as $file ) {
+				if ( pathinfo( $file, PATHINFO_EXTENSION ) === 'svg' ) {
+					$name     = explode( '.', $file )[0];
+					$filename = pathinfo( $file, PATHINFO_FILENAME );
+					$icon     = array(
+						'name'     => $name,
+						'filename' => $filename,
+						'icon'     => $file,
+					);
+					array_push( $this->svgs, $icon );
+				}
+			}
+
+			parent::__construct();
+		}
+
+		/**
+		 * Method that renders the field in the admin.
+		 *
+		 * @param array $field The field array.
+		 * @return void
+		 */
+		public function render_field( $field ) {
+			$input_icon = '' !== $field['value'] ? $field['value'] : $field['initial_value'];
+			$svg        = $this->path . $input_icon . '.svg';
+			?>
+			<div class="acf-svg-icon-picker">
+				<div class="acf-svg-icon-picker__img">
+					<?php
+					if ( file_exists( $svg ) ) {
+						$svg = $this->url . $input_icon . '.svg';
+						echo '<div class="acf-svg-icon-picker__svg">';
+						echo '<img src="' . esc_url( $svg ) . '" alt=""/>';
+						echo '</div>';
+					} else {
+						echo '<div class="acf-svg-icon-picker__svg">';
+						echo '<span class="acf-svg-icon-picker__svg--span">&plus;</span>';
+						echo '</div>';
+					}
+					?>
+					<input type="hidden" readonly name="<?php echo esc_attr( $field['name'] ); ?>" value="<?php echo esc_attr( $input_icon ); ?>" />
```

Comment:
Can maybe simplify this a little? Quick pseudo (untested code below) but def. not preventing a merge


```php
public function render_field( $field ) {
	$input_icon = '' !== $field['value'] ? $field['value'] : $field['initial_value'];
	$svg        = $this->path . $input_icon . '.svg';
	$svg_exists = file_exists( $svg );
	$svg_url    = esc_url($this->url . $input_icon . '.svg');
	$output     = $svg_exists ? "<img src='{$svg_url}' alt=''/>" : '<span class="acf-svg-icon-picker__svg--span">&plus;</span>';
	?>
	<div class="acf-svg-icon-picker">
		<div class="acf-svg-icon-picker__img">
		<div class="acf-svg-icon-picker__svg">
			<?php echo $output; ?>
		</div>
```

Reply:
Good suggestions, See my latest commit for an iteration on your code @mike-sheppard